### PR TITLE
FINERACT-1901 : Reject loan does not lead to NullPointer

### PR DESCRIPTION
**Pull request by Kengne-yvana**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by Kengne-yvana** [comment]
All i did was to add a try and catch exception to the readMappingByLoanId method in CreditBureauLoanProductMappingReadPlatformServiceImpl

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/service/CreditBureauLoanProductMappingReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -100,11 +101,15 @@ public Collection<CreditBureauLoanProductMappingData> readCreditBureauLoanProduc
...
         this.context.authenticatedUser();
+        try{
+            final CreditBureauLoanProductMapper rm = new CreditBureauLoanProductMapper();
+            final String sql = "select " + rm.schema() + " and cblp.loan_product_id=?";
 
-        final CreditBureauLoanProductMapper rm = new CreditBureauLoanProductMapper();
-        final String sql = "select " + rm.schema() + " and cblp.loan_product_id=?";
+            return this.jdbcTemplate.queryForObject(sql, rm, new Object[] { loanProductId }); // NOSONAR
+        }catch(EmptyResultDataAccessException e){
+            return  null;
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/service/CreditBureauLoanProductMappingReadPlatformServiceImpl.java:110]: I don't think this is the right way to do it. The API will respond with a null response body but it should simply signal an HTTP 404 representing a resource not found situation.

**Review by galovics — CHANGES_REQUESTED** [review]
Code is not formatted properly. There's no test to cover the changes and the implementation is not the right way to do it. Please take care of it.

For formatting, use ./gradlew spotlessApply

**Comment by Kengne-yvana** [comment]
Ok

**Pull request closed without merging**